### PR TITLE
promethues_scaler: add url in error information

### PR DESCRIPTION
**Pull request by surki**
Without the query its hard to debug which query is causing the error when there are hundreds of scalers.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -51,6 +51,7 @@ To learn more about our roadmap, we recommend reading [this document](ROADMAP.md
 - **AWS SQS Queue Scaler:** Support for scaling to include in-flight messages. ([#3133](https://github.com/kedacore/keda/issues/3133))
 - **GCP Stackdriver Scaler:** Added aggregation parameters ([#3008](https://github.com/kedacore/keda/issues/3008))
 - **Prometheus Scaler:** Add ignoreNullValues to return error when prometheus return null in values ([#3065](https://github.com/kedacore/keda/issues/3065))
+- **Prometheus Scaler:** Improve error reporting, include the failed query in error message ([#3406](https://github.com/kedacore/keda/pull/3406))
```
Comment (JorTurFer) [inline CHANGELOG.md:54]: I think that in this change, updating the changelog it's not needed because it's s tiny change. Could you undo this change?

Comment (zroubalik) [reply CHANGELOG.md:54]: I think it is okay, maybe just put it into `Other` section.

Comment (surki) [reply CHANGELOG.md]: Moved it under "Other" section, thanks.

**Comment by zroubalik** [comment]
I wonder whether it is not a privacy problem to expose the full url in the logs.  Let me get back to you next week.

**Comment by tomkerkhove** [comment]
> I wonder whether it is not a privacy problem to expose the full url in the logs. Let me get back to you next week.

I don't think it hurts; if you have access to the logs then it's fair to say you are a cluster admin; no? I do get the concern though.

**Comment by JorTurFer** [comment]
> I don't think it hurts; if you have access to the logs then it's fair to say you are a cluster admin; no? I do get the concern though.

I wouldn't assume that people with access to logs are admins... I think that the url itself is not a secret (at this point in code, that variable contains only the url, not the url with user and password) but maybe I'm wrong

**Comment by zroubalik** [comment]
Would following feature help?
https://github.com/kedacore/keda/issues/3419

**Comment by surki** [comment]
> Would following feature help? #3419

yes, that should be good enough (as long as full name is there, namespace + name)

I will close this, thanks.
